.travis.yml

```diff
@@ -20,7 +20,13 @@ aliases:
   - &run-core-and-ext-tests |
     set -e
     jdk="$(ext/travisci/jdk-from-spec "$PDB_TEST")"
-    if test "$(uname)" = Linux; then jdk_switcher use "$jdk"; fi
+    jdkver="${jdk##*jdk}"
+    if test "$(uname)" = Linux -a "$jdkver" -lt 9 ; then
+        jdk_switcher use "$jdk";
+    elif test "$(uname)" = Linux; then
+        export JAVA_HOME="$(ext/bin/require-jdk "$jdk" ext/travisci/local)"
+        export PATH="$JAVA_HOME/bin:$PATH"
+    fi
```

Comment:
I'm inclined to just assert that what require-jdk does is create a valid JAVA_HOME in the specified dir.  Then we can avoid needing/caring-about the output. i.e.:

    ext/bin/require-jdk "$jdk" ext/travisci/local
    export JAVA_HOME="$(pwd)/ext/travisci/local"
    export PATH="$JAVA_HOME/bin:$PATH"

And then we also wouldn't need the tail at the end of require-jdk, etc.